### PR TITLE
engine: open web browser on first Tiltfile load

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: internal/engine/upper.go [hunk 1]
```diff
@@ -953,3 +956,8 @@ func handleTiltfileLogAction(ctx context.Context, state *store.EngineState, acti
 	state.CurrentTiltfileBuild.Log = model.AppendLog(state.CurrentTiltfileBuild.Log, action, state.LogTimestamps)
 	state.TiltfileCombinedLog = model.AppendLog(state.TiltfileCombinedLog, action, state.LogTimestamps)
 }
+
+func handleOpenWebUIAction(ctx context.Context, state *store.EngineState, action OpenWebUIAction) {
+	state.OpenedWebUIOnStart = true
+	_ = browser.OpenURL(action.WebURL.String())
```
Comment (nicks) [inline internal/engine/upper.go]: please do not use reducers this way. the really are supposed to be pure and not have side-effects.

**Comment by nicks** [comment]
let's talk about this more in-person

**Comment by nicks** [comment]
ya, actions are explicitly not for things with side-effects like opening web browsers. 

this is not a design pattern i invented, see 
https://redux.js.org/basics/actions
https://redux.js.org/basics/reducers

that has pretty strong language discouraging people from doing this.

File: internal/engine/configs_controller.go [hunk 2]
```diff
@@ -89,7 +88,7 @@ func (cc *ConfigsController) OnChange(ctx context.Context, st store.RStore) {
 
 		loadCtx := logger.WithLogger(ctx, logger.NewLogger(logger.Get(ctx).Level(), multiWriter))
 
-		tlr, err := cc.tfl.Load(loadCtx, tiltfilePath, matching)
+		tlr, err := cc.tfl.Load(loadCtx, tiltfilePath, matching, !state.FirstTiltfileBuildCompleted)
```
[the remark this answers is not in the document]
Comment (jazzdan) [reply internal/engine/configs_controller.go]: Oh yeah, I was thinking this was still in the blessed territory of the reducer but it's not. Will do.

[the remark this answers is not in the document]
Comment (jazzdan) [reply internal/engine/configs_controller.go]: Actually on second thought, there are some situations where we don't want this behavior (tests, demos) and the configs controller doesn't know when it is being run in that way, where the Load callsites do know. So I think I'll leave it as it is.

**Pull request merged**